### DISCUSSION TITLE
DatePicker Input example

### DISCUSSION THREAD

**Discussion by mathieu-fournier** [iTwin/iTwinUI-react#674] [Show and tell]
Hi all o/, 
[Here's](https://codesandbox.io/s/itwinui-react-datepickerinput-vq3rmi?file=/src/App.tsx) a Data Picker Input I've made on codesandbox. 
![image](https://user-images.githubusercontent.com/82905333/168310785-096bf893-13c5-4229-8d89-a7d86f58b3d8.png)
Both selecting a date with the calendar and typing in the input are supported.
I haven't implemented the support for hours and minutes yet.
It support localization. In the example, I use the [library date-fns](https://date-fns.org/v2.28.0/docs/I18n). See the dateFormatter.ts file.

Also [here's](https://codesandbox.io/s/itwinui-react-labeleddatepickerinput-b289c3?file=/src/App.tsx) an example on how to add a Label and error Message to a DatePicker Input. 
![image](https://user-images.githubusercontent.com/82905333/168311431-a137399d-177a-4c6d-9291-1d39e8887086.png)

Disclaimer : It doesnt support all status and features, so these are only examples on which to build upon.

To follow the official development of these features, you can follow this thread : https://github.com/iTwin/iTwinUI/issues/272
![image](https://user-images.githubusercontent.com/82905333/168311731-72e0f78d-0b3d-4ff7-9c8a-ca4f0eb6202e.png)

Thanks (:

**Comment by veekeys** [comment]
Good job, @mathieu-fournier !
That could be a pretty good start for a generic date picker input component. If you are interested in contributing, feel free!
